src/app/devices/deviceContent/components/digitalTwin/digitalTwinInterfaces.tsx

```diff
@@ -7,25 +7,43 @@ import { useSelector, useDispatch } from 'react-redux';
 import { Label } from 'office-ui-fabric-react/lib/Label';
 import { CommandBar, ICommandBarItemProps } from 'office-ui-fabric-react/lib/CommandBar';
 import { DetailsList, IColumn, SelectionMode } from 'office-ui-fabric-react/lib/DetailsList';
+import { MessageBar, MessageBarType } from 'office-ui-fabric-react/lib/MessageBar';
+import { Announced } from 'office-ui-fabric-react/lib/Announced';
+import { Pivot, PivotItem } from 'office-ui-fabric-react/lib/Pivot';
 import { RouteComponentProps, NavLink } from 'react-router-dom';
-import { getDigitalTwinComponentNameAndIdsSelector, getDigitalTwinDcmNameSelector, getDigitalTwinInterfacePropertiesWrapperSelector } from '../../selectors';
+import { getDigitalTwinModelId,
+    getDigitalTwinSynchronizationStatusSelector,
+    ComponentAndInterfaceId,
+    getComponentNameAndInterfaceIdArraySelector,
+    getModelDefinitionSyncStatusSelector,
+    getModelDefinitionWithSourceSelector } from '../../selectors';
 import { ROUTE_PARTS, ROUTE_PARAMS } from '../../../../constants/routes';
 import { getDeviceIdFromQueryString } from '../../../../shared/utils/queryStringHelper';
 import { useLocalizationContext } from '../../../../shared/contexts/localizationContext';
 import { ResourceKeys } from '../../../../../localization/resourceKeys';
 import { SynchronizationStatus } from '../../../../api/models/synchronizationStatus';
-import { getDigitalTwinInterfacePropertiesAction } from '../../actions';
-import { HeaderView } from '../../../../shared/components/headerView';
-import MultiLineShimmer from '../../../../shared/components/multiLineShimmer';
+import { getDigitalTwinAction, getModelDefinitionAction } from '../../actions';
 import { REFRESH } from '../../../../constants/iconNames';
 import { LARGE_COLUMN_WIDTH } from '../../../../constants/columnWidth';
-import '../../../../css/_digitalTwinNav.scss';
+import { ModelDefinitionWithSource } from '../../../../api/models/modelDefinitionWithSource';
+import InterfaceNotFoundMessageBoxContainer from '../shared/interfaceNotFoundMessageBarContainer';
+import { ModelDefinitionSourceView } from '../shared/modelDefinitionSource';
+import MaskedCopyableTextFieldContainer from '../../../../shared/components/maskedCopyableTextFieldContainer';
+import { MonacoEditorView } from '../../../../shared/components/monacoEditor';
+import { HeaderView } from '../../../../shared/components/headerView';
+import MultiLineShimmer from '../../../../shared/components/multiLineShimmer';
+import { setSettingsVisibilityAction } from '../../../../settings/actions';
+import '../../../../css/_digitalTwinInterfaces.scss';
 
 export interface DigitalTwinInterfacesProps extends RouteComponentProps{
-    isLoading: boolean;
-    nameToIds: object;
-    dcm: string;
-    refresh: (deviceId: string) => void;
+    isDigitalTwinLoading: boolean;
```

Comment:
why not direct useSelector ?

Reply:
no you mean change line 237 from modelDefinitionSynchronizationStatus === SynchronizationStatus.working to creating another selector?